src/main/scala/uk/gov/nationalarchives/FileProcessor.scala

```diff
@@ -163,6 +166,29 @@ class FileProcessor(
     }
   }.sequence
 
+  private def createBagInfoJson(treMetadata: TREMetadata): IO[String] = {
+    val bagInfoIdFields =
+      List(
+        IdField("ConsignmentReference", treMetadata.parameters.TDR.`Internal-Sender-Identifier`),
+        IdField("UpstreamSystemReference", treMetadata.parameters.TRE.reference)
+      )
+    val bagInfo = BagInfo(
+      treMetadata.parameters.TDR.`Source-Organization`,
+      treMetadata.parameters.TDR.`Consignment-Export-Datetime`,
+      "TRE: FCL Parser workflow",
+      "Born Digital",
+      "FCL",
+      bagInfoIdFields
+    )
+
+    Stream
```

Comment:
Similar comment to the other line, you can replace all the streaming with 
```scala
uploadAsFile(bagInfo.asJson.printWith(Printer.noSpaces), "bag-info.json")
```
Again, I know I did it weird in the first place.